src/libmongoc/src/mongoc/mongoc-stream-tls-secure-channel.c

```diff
@@ -884,12 +883,14 @@ mongoc_stream_tls_secure_channel_handshake (mongoc_stream_t *stream,
       break;
    }
 
-
    *events = 0;
-   bson_set_error (error,
-                   MONGOC_ERROR_STREAM,
-                   MONGOC_ERROR_STREAM_SOCKET,
-                   "TLS handshake failed");
+
+   if (error && !error->code) {
```

Comment:
The caller, `mongoc_stream_tls_handshake`, may not have initialized `error`. I suggest adding:
```
if (error) {
    memset (error, 0, sizeof (bson_error_t));
}
```
To the start of this function.